packages/amplify_datastore/example/integration_test/utils/setup_utils.dart

```diff
@@ -34,6 +56,45 @@ Future<void> configureDataStore() async {
 ///
 /// see: https:///github.com/aws-amplify/amplify-android/issues/1464
 Future<void> clearDataStore() async {
-  await Future.delayed(Duration(milliseconds: 100));
+  await Future.delayed(Duration(seconds: 2));
   await Amplify.DataStore.clear();
 }
+
+/// Am async operator that starts DataStore API sync.
+/// It returns a Future which complets when [Amplify.Hub] receives the ready
+/// event on [HubChannel.DataStore], or completes with an timeout error if
+/// the ready event hasn't been emitted with in 2 minutes.
+class DataStoreStarter {
+  final Completer _completer = Completer();
+  late StreamSubscription hubSubscription;
+
+  Future<void> startDataStore() async {
+    hubSubscription = Amplify.Hub.listen([HubChannel.DataStore], (event) {
+      if (event.eventName == 'ready') {
+        print(
+            '🎉🎉🎉DataStore is ready to start running test suites with API sync enabled.🎉🎉🎉');
+        hubSubscription.cancel();
+        _completer.complete();
```

Comment:
Is this expected to be done before/after `Amplify.Datastore.start()`? Appears should be after? It's a little confusing to me, appears that we expect to start listening to hub, start datastore, then hub completes later.

Reply:
Hub gets ready in general for all possible events from any category.
So we can setup the listener to hub even before starting DataStore to ensure the even can be always captured.

In this implementation, `_completer.complete();` is invoked after `Amplify.DataStore.start()` completed. As the `start` API only starts storage engine and sync engine, as long as the starting completes, it returns, and not ensure if the ready event is dispatched or not.

Hope this explanation helps understanding.

Reply:
thanks